src/rgw/rgw_cache.cc

```diff
@@ -352,3 +362,431 @@ ObjectCache::~ObjectCache()
   }
 }
 
+int cacheAioWriteRequest::create_io(bufferlist& bl, unsigned int len, string oid) {
+  std::string location = cct->_conf->rgw_datacache_persistent_path + oid;
```

Comment:
It won't change from request to request. I'll add the change to save the cache path after adding a trailing slash to it (if it's missing) in a separate PR once this one merged, if that's okay.

Reply:
of course

Reply:
checked, it's is not trivial - the `oid`varies for example:
`...2__multipart_foo.2~bw1mM-Y9hULvgL4T8p7Kfr4En9wnXlR.1` to
`...2__shadow_foo.2~bw1mM-Y9hULvgL4T8p7Kfr4En9wnXlR.1_1`
will investigate further

Reply:
> is this able to change from request to request? otherwise, should we save the persistent path with the cache to avoid reading the location on each one?

done